### PR TITLE
[KYUUBI #6147][TRINO] Use the overlay conf as session conf

### LINKED ISSUE #6147: [Bug][TRINO] Trino engine use the engine level conf as session level conf
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

In TrinoSessionImpl, it's wrong consider sessionManager.getConf as session level conf. Actually, it's engine level conf.

https://github.com/apache/kyuubi/blob/6fc7552ef05c83cfb8a2cd56d80c02c6e8a033c7/externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala#L52

We should use the passed arg conf as session level conf.

https://github.com/apache/kyuubi/blob/6fc7552ef05c83cfb8a2cd56d80c02c6e8a033c7/externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala#L43-L50

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zwangsheng**
# :mag: Description
## Issue References 🔗

This pull request fixes #6147

## Describe Your Solution 🔧

Overwriting the engine conf with the passed conf consider as the full session conf.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests
None

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 1]
```diff
@@ -49,7 +49,11 @@ class TrinoSessionImpl(
     sessionManager: SessionManager)
   extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
 
-  val sessionConf: KyuubiConf = sessionManager.getConf
+  val sessionConf: KyuubiConf = {
+    val engineConf = sessionManager.getConf
+    conf.foreach(kv => engineConf.set(kv._1, kv._2))
```
Comment (cxzl25) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:54]: Will this not pollute the engine conf?

Comment (zwangsheng) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:54]: thanks for catch, using `sessionManager.getConf.clone` instead.

Comment (pan3793) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:54]: @cxzl25 seems we lack a pre-assembled session conf in AbstractSession, we need some major refactor to simplify the current conf evaluation, I roughly remember you have raised a similar idea previously.

**Review by pan3793 — APPROVED** [review]
LGTM, I think we can adopt this ad-hoc fix and reconsider the https://github.com/apache/kyuubi/commit/d0f479831a5fca72dc10bf97f2627b05ac07946b 's idea later

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 2]
```diff
@@ -49,7 +49,11 @@ class TrinoSessionImpl(
     sessionManager: SessionManager)
   extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
 
-  val sessionConf: KyuubiConf = sessionManager.getConf
+  val sessionConf: KyuubiConf = {
+    val engineConf = sessionManager.getConf.clone
+    conf.foreach(kv => engineConf.set(kv._1, kv._2))
```
Comment (pan3793) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:54]:
```suggestion
    conf.foreach { case (k, v) => engineConf.set(k, v) }
```

**Comment by pan3793** [comment]
Thanks, merged to master/1.8.2

**Pull request closed without merging**